control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2703,14 +2703,14 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 
 		cfg := manifests.OpenStackProviderConfig(hcp.Namespace)
 		if _, err := createOrUpdate(ctx, r, cfg, func() error {
-			return openstack.ReconcileCloudConfigConfigMap(hcp.Spec.Platform.OpenStack.ExternalNetwork.ID, cfg, hcp.Spec.Platform.OpenStack.IdentityRef.CloudName, credentialsSecret, caCertData)
+			return openstack.ReconcileCloudConfigConfigMap(hcp.Spec.Platform.OpenStack.ExternalNetwork.ID, cfg, hcp.Spec.Platform.OpenStack.IdentityRef.CloudName, credentialsSecret, caCertData, hcp.Spec.Networking.MachineNetwork)
```

Comment:
This has been done in v2:

https://github.com/openshift/hypershift/blob/17a958fe04edb8ca2ce45c84553ca95a773246df/control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/openstack/config.go#L73

Which is supposed to be transitioned in 4.19. I'm ok to make that change but this will involve other changes not related to this PR. WDYT?

Reply:
This has been done in hostedcontrolplane v2 I believe:

https://github.com/openshift/hypershift/blob/17a958fe04edb8ca2ce45c84553ca95a773246df/control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/openstack/config.go#L73

Which is supposed to be transitioned in 4.19. I'm ok to make that change but this will involve other changes not related to this PR. WDYT?

Reply:
That sounds reasonable to me

Reply:
I ended up doing it on https://github.com/openshift/hypershift/pull/5091 because of https://issues.redhat.com/browse/OCPBUGS-44375